### PR TITLE
Fix double scrollbars on product editor page

### LINKED ISSUE #38149: Product Block Editor: Fix dual scrollbars
**Description**

We're seeing a dual scrollbar when viewing the product block editor:

image

This stems from the fact that the global sidebar is actually not fixed, but the interface skeleton container is. It's then creating a scrollable body container where our form lives. This could be changed to make the sidebar fixed (as it is with the previous react and legacy editors via the `sticky-*` body class), but that would be diverging from the blocks way.

**Acceptance criteria**

- [ ] Only the relevant scrollbar is visible when viewing the editor.
- [ ] The global sidebar still works as expected.

### PR DESCRIPTION
**Pull request by joshuatf**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR fixes the double scrollbars in the page.  In order to do this, this PR does a few things:

* Refactors the `Layout` component to a functional component (to allow easier use of hooks)
* Creates a hook that helps create a slug from a route path and adds this class to the body to allow styling various elements on a page
* Styles the interface skeleton without fixing its position to the screen

Closes #38149 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
2. Under Features tab make sure to enable product-block-editor
3. Then visit `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. Reduce your viewport/browser height so you can scroll the page
5. Make sure that only one scrollbar is shown when scrolling the product editor
6. Narrow your viewport and make sure the header stays the correct width and location on various screen sizes (note that padding of the editor on mobile devices will be adjusted in a separate issue)
7. Open your console
8. Run `localStorage.setItem( 'debug', 'wc-admin:tracks' );` in the console.
9. Navigate to various WooCommerce Admin pages (not legacy pages), such as Analytics pages, Home, or the new Product editing experience
10. Make sure that `wcadmin_page_view` event is correctly triggered on each page

### CODE REVIEW

File: packages/js/product-editor/src/components/editor/style.scss [hunk 1]
```diff
@@ -1,18 +1,45 @@
 @import '@wordpress/interface/src/style.scss';
 
-.interface-interface-skeleton {
-	@include breakpoint( '<782px' ) {
-		top: $adminbar-height-mobile;
+.woocommerce-admin-page__product_product-id {
```
Comment (louwie17) [inline packages/js/product-editor/src/components/editor/style.scss]: I don't think we should be adding a WooCommerce Admin specific page class to the css of the editor within a package. If this CSS is specific to the product page it should be moved to the `product-page.scss` sheet.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Nice work! I like the creation of the `usePageClasses` hook 👌 
I did notice the fix only works for existing products and not on the create new page:
Screenshot 2023-05-15 at 11 03 40 AM

I did leave an inline comment about moving some of that css to WooCommerce Admin instead of the Editor component.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Nice work! I like the creation of the `usePageClasses` hook 👌 
I did notice the fix only works for existing products and not on the create new page:
Screenshot 2023-05-15 at 11 03 40 AM

I did leave an inline comment about moving some of that css to WooCommerce Admin instead of the Editor component.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Nice work! I like the creation of the `usePageClasses` hook 👌 
I did notice the fix only works for existing products and not on the create new page:
Screenshot 2023-05-15 at 11 03 40 AM

I did leave an inline comment about moving some of that css to WooCommerce Admin instead of the Editor component.

**Comment by joshuatf** [comment]
@louwie17 Good catch!  And agreed.  I moved the product page styles to our client.  This required a quick rename of the (old-new) existing stylesheet.

**Review by louwie17 — APPROVED** [review]
Nice work, and thanks for moving some of the wcAdmin specific css to the client. This tested well 🚀

**Review by louwie17 — DISMISSED** [review]
Nice work, and thanks for moving some of the wcAdmin specific css to the client. This tested well 🚀 

I guess just requires some lint error fixing.

**Comment by joshuatf** [comment]
Hey @louwie17, had to fix up a failing test.  Could you give this another look?

**Review by louwie17 — APPROVED** [review]
Thanks for fixing the tests, LGTM 🚀

**Pull request merged**